### PR TITLE
refactor(CLI): `routes` command and associated tests

### PR DESCRIPTION
**Pull request by provinzkraut**
Refactor the CLI `routes` command and associated tests to reduce complexity. 
This should also fix the failing tests on Windows.

https://sonarcloud.io/project/issues?id=litestar-org_litestar&open=AYzgW0Tm5TPoZDnT_BAi

### CODE REVIEW

**Review by guacs — APPROVED** [review]
Nice :)

File: tests/unit/test_cli/test_core_commands.py [hunk 1]
```diff
@@ -327,12 +327,24 @@ def test_run_command_with_server_lifespan_plugin(
...
 
 @pytest.mark.parametrize(
-    "app_content, schema_enabled, exclude_pattern_list",
+    "app_content, schema_enabled, exclude_pattern_list, expected_result_routes_count",
     [
-        (APP_FILE_CONTENT_ROUTES_EXAMPLE, False, ()),
-        (APP_FILE_CONTENT_ROUTES_EXAMPLE, False, ("/foo", "/destroy/.*", "/java", "/haskell")),
-        (APP_FILE_CONTENT_ROUTES_EXAMPLE, True, ()),
-        (APP_FILE_CONTENT_ROUTES_EXAMPLE, True, ("/foo", "/destroy/.*", "/java", "/haskell")),
+        pytest.param(APP_FILE_CONTENT_ROUTES_EXAMPLE, False, (), 3, id="schema-enabled_no-exclude"),
```
Comment (peterschutt) [inline tests/unit/test_cli/test_core_commands.py:332]: @provinzkraut are the IDs backward here, i.e., this should be `schema-disabled_no-exclude`?

Comment (provinzkraut) [reply tests/unit/test_cli/test_core_commands.py:332]: Seems like it 😬

Comment (peterschutt) [reply tests/unit/test_cli/test_core_commands.py:332]: yeh sweet - np I just wanted to make sure I wasn't missing anything. I've updated in the OpenAPI plugin PR.

**Pull request merged**